### PR TITLE
fix: card header line clamp

### PR DESCRIPTION
**Pull request by el**
## Changes
SCR-20230206-pik

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/lib/dateFormat.ts [hunk 1]
```diff
@@ -22,11 +29,12 @@ export function postDateFormat(
...
 
-  return date.toLocaleString('en-US', {
+  const options: Intl.DateTimeFormatOptions = {
     month: 'short',
     day: '2-digit',
-    year: 'numeric',
-  });
+  };
+  if (!isSameYear(date, now)) options.year = 'numeric';
+  return date.toLocaleString('en-US', options);
```
Comment (rebelchris) [inline packages/shared/src/lib/dateFormat.ts:37]: Maybe just double check product and design are ok with this :)

**Review by rebelchris — APPROVED** [review]
Happy with this, we just need to verify it before merging to avoid confusion.

File: packages/shared/src/components/cards/SharedPostCardHeader.tsx [hunk 2]
```diff
@@ -30,15 +30,19 @@ export const SharedPostCardHeader = ({
           size="xsmall"
         />
       </div>
-      <div className="flex flex-col flex-grow typo-footnote">
-        <span className="font-bold">{author.name}</span>
+      <div className="flex flex-col flex-1 flex-grow ml-2 typo-footnote">
+        <span className="font-bold line-clamp-2">{author.name}</span>
         <PostMetadata
-          className="text-theme-label-secondary"
+          className="line-clamp-1 text-theme-label-secondary !flex"
```
Comment (sshanzel) [inline packages/shared/src/components/cards/SharedPostCardHeader.tsx:36]: No way! I never knew the usage of `!` prefix. Wow, I am learning a lot today 🤣

**Review by sshanzel — COMMENTED** [review]
Got nothing other than what Chris has mentioned. Nice work! I learned something today! 🤩

File: packages/shared/src/components/cards/SharedPostCardHeader.tsx (hunk 2 above)
Comment (el) [reply packages/shared/src/components/cards/SharedPostCardHeader.tsx:36]: Haha. Yeah can be useful but should not be overused 😃

File: packages/shared/src/lib/dateFormat.ts (hunk 1 above)
Comment (el) [reply packages/shared/src/lib/dateFormat.ts:37]: This is actually according to the design

SCR-20230206-vlq

**Pull request merged**